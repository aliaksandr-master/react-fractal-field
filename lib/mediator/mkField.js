export const mkMeta = () => ({
  valid: true,
  touched: false,

  active: false,
  activated: false,

  submitted: false,
  submitting: false,
  submitFailed: false,
  submitSuccess: false,
  submittedTimes: 0,

  error: null,
  errorsIn: [],
  hasException: false,
  hasFormatException: false,
  hasNormalizeException: false,
  hasValidateException: false
});

export default ({
  id,
  complexValueType = null,
  parentID = null,
  autoClean = false,
  shareMeta = true,
  shareValue = true,
  value = undefined,
  name = null
}) => ({
  // permanent
  id,
  parentID,
  autoClean,
  shareMeta,
  shareValue,
  complexValueType, // if not null component might have named children

  // free transform
  name,

  value,

  meta: mkMeta(),

  children: [],
  version: 0,
  metaVersion: 0,
  valueVersion: 0
});
