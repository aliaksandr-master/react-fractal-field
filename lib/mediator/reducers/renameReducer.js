import recalcValue from '../recalcValue';



export default () => (state, { fieldID, payload: name = null }) => {
  const field = state.fields[fieldID];

  if (field.name === name) {
    return state;
  }

  state = {
    ...state,
    fields: {
      ...state.fields,
      [fieldID]: { ...field, name }
    }
  };

  return recalcValue(state, field.parentID, field.id);
};
