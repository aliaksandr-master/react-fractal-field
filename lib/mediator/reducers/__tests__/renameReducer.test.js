/*eslint-env jest*/
import renameReducer from '../renameReducer';
import mkField from '../../mkField';


test('renameReducer', () => {
  expect(renameReducer()({}, { type: 'some', fieldID: 123 })).toMatchSnapshot();

  expect(renameReducer()({ [123]: mkField({ id: 123 }) }, { type: 'some', fieldID: 123, payload: undefined })).toMatchSnapshot();

  expect(renameReducer()({ [123]: mkField({ id: 123 }) }, { type: 'some', fieldID: 123, payload: '234' })).toMatchSnapshot();

  expect(renameReducer()({ [123]: mkField({ id: 123, name: '234' }) }, { type: 'some', fieldID: 123, payload: '234' })).toMatchSnapshot();
});
