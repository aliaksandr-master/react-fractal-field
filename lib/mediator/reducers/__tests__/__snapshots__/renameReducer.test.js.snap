// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renameReducer 1`] = `Object {}`;

exports[`renameReducer 2`] = `
Object {
  "123": Object {
    "autoClean": false,
    "children": Array [],
    "complexValueType": null,
    "id": 123,
    "meta": Object {
      "activated": false,
      "active": false,
      "error": null,
      "errorsIn": Array [],
      "hasException": false,
      "hasFormatException": false,
      "hasNormalizeException": false,
      "hasValidateException": false,
      "submitFailed": false,
      "submitSuccess": false,
      "submitted": false,
      "submittedTimes": 0,
      "submitting": false,
      "touched": false,
      "valid": true,
    },
    "metaVersion": 0,
    "name": null,
    "parentID": null,
    "shareMeta": true,
    "shareValue": true,
    "value": undefined,
    "valueVersion": 0,
    "version": 0,
  },
}
`;

exports[`renameReducer 3`] = `
Object {
  "123": Object {
    "autoClean": false,
    "children": Array [],
    "complexValueType": null,
    "id": 123,
    "meta": Object {
      "activated": false,
      "active": false,
      "error": null,
      "errorsIn": Array [],
      "hasException": false,
      "hasFormatException": false,
      "hasNormalizeException": false,
      "hasValidateException": false,
      "submitFailed": false,
      "submitSuccess": false,
      "submitted": false,
      "submittedTimes": 0,
      "submitting": false,
      "touched": false,
      "valid": true,
    },
    "metaVersion": 0,
    "name": "234",
    "parentID": null,
    "shareMeta": true,
    "shareValue": true,
    "value": undefined,
    "valueVersion": 0,
    "version": 1,
  },
}
`;

exports[`renameReducer 4`] = `
Object {
  "123": Object {
    "autoClean": false,
    "children": Array [],
    "complexValueType": null,
    "id": 123,
    "meta": Object {
      "activated": false,
      "active": false,
      "error": null,
      "errorsIn": Array [],
      "hasException": false,
      "hasFormatException": false,
      "hasNormalizeException": false,
      "hasValidateException": false,
      "submitFailed": false,
      "submitSuccess": false,
      "submitted": false,
      "submittedTimes": 0,
      "submitting": false,
      "touched": false,
      "valid": true,
    },
    "metaVersion": 0,
    "name": "234",
    "parentID": null,
    "shareMeta": true,
    "shareValue": true,
    "value": undefined,
    "valueVersion": 0,
    "version": 0,
  },
}
`;
